leveldb/leveldb.go

```diff
@@ -0,0 +1,101 @@
+package leveldb
+
+import (
+	"fmt"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+// Manager struct
+type Manager struct {
+	db *leveldb.DB
+}
+
+// Item struct
+type Item struct {
+	Key []byte
+	Value []byte
+}
+
+// IManager - Manager interface
+type IManager interface {
+	Close()
+	GetAllItems() (items []Item, err error)
+	AddNewItem(item Item) (status int, err error)
+	FindItemByKey(key []byte) (item *Item, err error) 
+}
+
+// Error - custom leveldb error struct
+type Error struct {
+	Key []byte
+	err error
+}
+
+// NewErr - create new custom error instance
+func NewErr(key []byte, err error) error {
+	return &Error{
+		Key: key,
+		err: err,
+	}
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error at Key: %s\nMessage:\n    %s", e.Key, e.err.Error())
+}
+
+// Close database connection,
+// (*Manager).db cannot be used after this function called.
+func (mgr *Manager) Close() {
+	mgr.db.Close()
+}
+
+// FindItemByKey - Return Item object with key and value.
+func (mgr *Manager) FindItemByKey(key []byte) (item *Item, err error) {
+	var value []byte
+	value, err = mgr.db.Get(key, nil)
+	if err != nil {
+		err = NewErr(key, err)
+		return nil, err
+	}
+	item = &Item {
+		Key: key,
+		Value: value,
+	}
+	return item, nil
+}
+
+// AddNewItem add new item into database
+func (mgr *Manager) AddNewItem(item Item) (status int, err error) {
```

Comment:
**status** looks like doesn't necessary
```go
func (mgr *Manager) AddNewItem(item Item) error
```
Is good enough, beside we need method like:
```go
func (mgr *Manager) Set(key []byte, value []byte) error
func (mgr *Manager) Get(key[]) (err error, value []byte)
```
since it would be use more usual than AddNewItem.